x/tx/signing/aminojson/encoder.go

```diff
@@ -81,6 +81,22 @@ func nullSliceAsEmptyEncoder(enc *Encoder, v protoreflect.Value, w io.Writer) er
 	}
 }
 
+// cosmosBytesAsString replicates the behavior at:
+// https://github.com/CosmWasm/wasmd/blob/08567ff20e372e4f4204a91ca64a371538742bed/x/wasm/types/tx.go#L20-L22
+func cosmosBytesAsString(_ *Encoder, v protoreflect.Value, w io.Writer) error {
+	switch bz := v.Interface().(type) {
```

Comment:
can this entire block be replaced with
```golang
_, err = w.Write(string(bz))
return err
```

Reply:
`Write` does not accept `string` , only `[]byte` . I tried to remove the cast and it seems to work fine in wasmd ( msg is not base64 encoded)
then the function would look like this:
```
func cosmosBytesAsString(_ *Encoder, v protoreflect.Value, w io.Writer) error {
	switch bz := v.Interface().(type) {
	case []byte:
		_, err := w.Write(bz)
		return err
	default:
		return fmt.Errorf("unsupported type %T", bz)
	}
}
```
Do you agree?

Reply:
@kocubinski @julienrbrt I did some tests and the output between the two implementations is the same.
Personally I would prefer the first implementation because it is more clear and the implementation matches the one in wasmd. On the other side, the second option is much shorter so it is nice as well.
Please let me know how to proceed so I will do the needed changes. Thank you!

Reply:
Sorry for the delay, please feel free to ping me in Slack if I'm not responsive on this PR.  

I find below (the first implementation) confusing:
```golang
		blob, err := json.RawMessage(bz).MarshalJSON()
		if err != nil {
			return err
		}
		_, err = w.Write(blob)
```
There is mention of json, but `bz` need not even be valid json!  the effect of this code is just to write bytes as a literal, with the exception that when the byte slice is nil the string literal "null" is printed.

```golang
// RawMessage is a raw encoded JSON value.
// It implements [Marshaler] and [Unmarshaler] and can
// be used to delay JSON decoding or precompute a JSON encoding.
type RawMessage []byte

// MarshalJSON returns m as the JSON encoding of m.
func (m RawMessage) MarshalJSON() ([]byte, error) {
	if m == nil {
		return []byte("null"), nil
	}
	return m, nil
}
```

I don't want to be overly pedantic here either, since as you say either will produce the same results.  Given that I'll approve as is.

Reply:
> There is mention of json, but bz need not even be valid json! the effect of this code is just to write bytes as a literal

The code you are referencing requires bytes to be valid JSON which is enforced in a different part of the code:

```go
func (r *RawContractMessage) ValidateBasic() error {
	if r == nil {
		return ErrEmpty
	}
	if !json.Valid(*r) {
		return ErrInvalid
	}
	return nil
}
```

So yeah, it is technically possible for a `RawContractMessage` instance to contain invalid JSON. But every valid usage of that type must perform this check and can then make assumptions on JSON validity later on. I with this was guaranteed at type level instead of app logic levels but this is not how json.RawMessage work and creating a custom "valid JSON" type seemed to be a bit overkill and not idiomatic.